receiver/windowsperfcountersreceiver/windowsperfcounters_scraper.go

```diff
@@ -44,41 +44,47 @@ type PerfCounterScraper interface {
 
 // scraper is the type that scrapes various host metrics.
 type scraper struct {
-	cfg      *Config
-	logger   *zap.Logger
-	counters []PerfCounterScraper
+	cfg                     *Config
+	settings                component.TelemetrySettings
+	counters                []PerfCounterMetrics
+	undefinedMetricCounters []PerfCounterScraper
 }
 
-func newScraper(cfg *Config, logger *zap.Logger) (*scraper, error) {
-	if err := cfg.Validate(); err != nil {
-		return nil, err
-	}
+type PerfCounterMetrics struct {
+	CounterScraper PerfCounterScraper
+	Attributes     map[string]string
+	Metric         string
+}
 
-	s := &scraper{cfg: cfg, logger: logger}
-	return s, nil
+func newScraper(cfg *Config, settings component.TelemetrySettings) *scraper {
+	s := &scraper{cfg: cfg, settings: settings}
+	return s
 }
 
 func (s *scraper) start(context.Context, component.Host) error {
 	var errs error
 
 	for _, perfCounterCfg := range s.cfg.PerfCounters {
 		for _, instance := range perfCounterCfg.instances() {
-			for _, counterName := range perfCounterCfg.Counters {
-				counterPath := counterPath(perfCounterCfg.Object, instance, counterName)
+			for _, counterCfg := range perfCounterCfg.Counters {
+				counterPath := counterPath(perfCounterCfg.Object, instance, counterCfg.Name)
 
 				c, err := pdh.NewPerfCounter(counterPath, true)
 				if err != nil {
 					errs = multierr.Append(errs, fmt.Errorf("counter %v: %w", counterPath, err))
 				} else {
-					s.counters = append(s.counters, c)
+					if counterCfg.Metric == "" {
+						s.undefinedMetricCounters = append(s.undefinedMetricCounters, c)
+					}
+					s.counters = append(s.counters, PerfCounterMetrics{CounterScraper: c, Metric: counterCfg.Metric, Attributes: counterCfg.Attributes})
```

Comment:
Do we only want to add to one or the other? Looks like we're adding to both when the metric is not specified.

Reply:
Whoops yes, added a continue to that check